### ISSUE TITLE
[REQ] Support Jakarta EE 9

### ISSUE THREAD

**Issue by ferblaca** [mockrunner/mockrunner#83]
With the upgrade to Spring Boot 3 and the adoption of Jakarta EE9, Jakarta EE 9 a new top-level jakarta package, replacing EE 8’s javax top-level package.

For example, the current `com.mockrunner.mock.jms.MockSession` class is using the `javax.jms.` packages instead of `jakarta.jms.`.

Is there provision for a release using Jakarta EE9?

**Comment by ferblaca** [comment]
@steinarb Is this repository maintained? If such a contribution is made, would it at least be included?

**Comment by steinarb** [comment]
> [@steinarb](https://github.com/steinarb) Is this repository maintained? If such a contribution is made, would it at least be included?

@ferblaca I'm not the primary maintainer of this project, so I can't say. @cemartins is AFAIK the one to ask.

I am, however, a heavy user of the mockrunner-servlet mocks and I just ran into https://github.com/mockrunner/mockrunner/issues/90

Which made me think about the fact that I will have to make the transition from javax.servlet to jakarta.servlet soon and what will I do with all of my unit tests using mockrunner then?

And searching for "jakarta" in issues landed me on this issue, which went under my radar in 2023

I use MockHttpServletRequest and MockHttpServletResponse instead of mockito mocks because it is easier to use and populate with real connected data.  So I will either need to do something in the tests (replace with mockito mocks) or ensure there is a mockrunner with classes from mockrunner-servlet.

So what would be the best way to move to jakarta.servlet and jakarta.inject (which AFAICT are the only javax-packages replaced with jakarta-packages that are in mockrunner)?

1. A clean break with a new major version, supporting jakarta.x in 3.x and javax.* if necessary, in further 2.0.x versions?
2. Duplicate existing classes in a different packages, containing "jakarta" somewhere in the package name, in the mockrunner-jms-spring and mockrunner-servlet jar files?
3. Copy the mockrunner-jms-spring and mockrunner-servlet maven modules to new modules mockrunner-jms-spring-jakarta or mockrunner-servlet-jakarta and put all files in packages with jakarta added somewhere?

**Comment by cemartins** [comment]
Yeah, I needed Mockrunner as maven dependencies back in 2012 or 13... long time ago. The original developers didn't want to port it to maven, so I did.   
@steinarb joined the project a little bit later, but also a long time ago already.

Nowadays I don't use Mockrunner at all and I confess that I also pay little to no attention to its maintenance. Thanks to. @steinarb it still moves along.

I created this Github "organization" called Mockrunner that has 2 repos and 5 members, three of which are owners:
- @ibbaa (initial developer)
- Me (guy who mavenized it)
- @steinarb - who I just made owner and who deserved to be owner a long time ago.

More members / contributors are very welcome, of course.

**Comment by steinarb** [comment]
>>>>> Carlos Martins ***@***.***>:

> cemartins left a comment (mockrunner/mockrunner#83)
> Yeah, I needed Mockrunner as maven dependencies back in 2012 or 13... long time ago. The original developers didn't want to port it to maven, so I did.   
> @steinarb joined the project a little bit later, but also a long time ago already.

> Nowadays I don't use Mockrunner at all and I confess that I also pay little to no attention to its maintenance. Thanks to. @steinarb it still moves along.

And I'm only using mockrunner-servlet myself. I know next to nothing
about the other modules.

I have found mockrunner-servlet useful for servlet api testing and it
has worked as well for me in 2026 as it did in 2016, until it no longer
did... https://github.com/mockrunner/mockrunner/issues/90

Also: I'm in the process of moving from javax.servlet to
jakarta.servlet... so it's either finding a solution for
mockrunner-servlet, or moving to mockito for everything.

> I created this Github "organization" called Mockrunner that has 2 repos and 5 members, three of which are owners:
> - @ibbaa (initial developer)
> - Me (guy who mavenized it)
> - @steinarb - who I just made owner and who deserved to be owner a long time ago.

Ok, if there are no objections I will roll a release with the 3 years of
stuff currently on main since the last release, plus a bump of
javax.servlet-api to the latest release made of javax.servlet-api (4.0.1
from April 2018), because of what I just encountered with shiro 2.2.0.
 https://github.com/mockrunner/mockrunner/issues/90

I am in the process of creating a PR for issue #90, and would much prefer
other eyes on the PR before merging.

It should be quite straightforward: just a maven dependency change and
implementing all of the missing methods resulting from the upgrade.

But it was many methods that needed a skeleton implementaton.

After that... I don't know...? A jakarta.* release of some kind...?

Does anyone have any thoughts on that?

**Comment by cemartins** [comment]
No objections at all. Go for it. And I’ll be happy to check your PR. let me know when it’s there 👍CheersCarlosSent from my iPadOn 22 May 2026, at 19:34, Steinar Bang ***@***.***> wrote:﻿steinarb left a comment (mockrunner/mockrunner#83)
>>>>> Carlos Martins ***@***.***>:

> cemartins left a comment (mockrunner/mockrunner#83)
> Yeah, I needed Mockrunner as maven dependencies back in 2012 or 13... long time ago. The original developers didn't want to port it to maven, so I did.   
> @steinarb joined the project a little bit later, but also a long time ago already.

> Nowadays I don't use Mockrunner at all and I confess that I also pay little to no attention to its maintenance. Thanks to. @steinarb it still moves along.

And I'm only using mockrunner-servlet myself. I know next to nothing
about the other modules.

I have found mockrunner-servlet useful for servlet api testing and it
has worked as well for me in 2026 as it did in 2016, until it no longer
did... https://github.com/mockrunner/mockrunner/issues/90

Also: I'm in the process of moving from javax.servlet to
jakarta.servlet... so it's either finding a solution for
mockrunner-servlet, or moving to mockito for everything.

> I created this Github "organization" called Mockrunner that has 2 repos and 5 members, three of which are owners:
> - @ibbaa (initial developer)
> - Me (guy who mavenized it)
> - @steinarb - who I just made owner and who deserved to be owner a long time ago.

Ok, if there are no objections I will roll a release with the 3 years of
stuff currently on main since the last release, plus a bump of
javax.servlet-api to the latest release made of javax.servlet-api (4.0.1
from April 2020), because of what I just encountered with shiro 2.2.0.
 https://github.com/mockrunner/mockrunner/issues/90

I am in the process of creating a PR for issue #90, and would much prefer
other eyes on the PR before merging.

It should be quite straightforward: just a maven dependency change and
implementing all of the missing methods resulting from the upgrade.

But it was many methods that needed a skeleton implementaton.

After that... I don't know...? A jakarta.* release of some kind...?

Does anyone have any thoughts on that?
